apps/etterlatte-hendelser-joark/src/main/kotlin/joarkhendelser/JoarkHendelseHandler.kt

```diff
@@ -40,38 +43,66 @@ class JoarkHendelseHandler(
 ) {
     private val logger: Logger = LoggerFactory.getLogger(JoarkHendelseHandler::class.java)
 
-    suspend fun haandterHendelse(hendelse: JournalfoeringHendelseRecord) {
-        val hendelseId = hendelse.hendelsesId
-        val journalpostId = hendelse.journalpostId
-        val temaNytt = hendelse.temaNytt
-
-        if (!hendelse.erTemaEtterlatte()) {
-            logger.debug("Hendelse (id=${hendelse.hendelsesId}) har tema ${hendelse.temaNytt} og håndteres ikke")
-            return // Avbryter behandling
-        }
-
-        logger.info(
-            "Starter behandling av hendelse (id=$hendelseId, journalpostId=$journalpostId, tema=$temaNytt)",
-        )
-
+    private suspend fun hentJournalpost(journalpostId: Long): Journalpost {
         val response = safKlient.hentJournalpost(journalpostId)
 
         val journalpost =
             if (response.errors.isNullOrEmpty()) {
                 response.data?.journalpost
             } else {
-                throw mapError(response.errors)
+                throw mapError(response.errors, journalpostId)
             }
 
         if (journalpost == null) {
             throw NullPointerException("Fant ingen journalpost med id=$journalpostId")
-        } else if (journalpost.erFerdigstilt()) {
+        }
+        return journalpost
+    }
+
+    private fun journalpostErFerdigstilt(journalpost: Journalpost): Boolean {
+        if (journalpost.erFerdigstilt()) {
             logger.info(
-                "Journalpost med id=$journalpostId er allerede ferdigstilt og tilknyttet sak (${journalpost.sak})",
+                "Journalpost med id=${journalpost.journalpostId} er allerede ferdigstilt og tilknyttet sak (${journalpost.sak})",
             )
+            return true
+        } else {
+            return false
+        }
+    }
+
+    private fun bytterFraEtterlatteTemaTilNoeAnnet(hendelse: JournalfoeringHendelseRecord): Boolean =
+        hendelse.erGammeltTemaEtterlatte() && !hendelse.erNyttTemaEtterlatte()
+
+    suspend fun haandterHendelse(hendelse: JournalfoeringHendelseRecord) {
+        val hendelseId = hendelse.hendelsesId
+        val journalpostId = hendelse.journalpostId
+        val temaNytt = hendelse.temaNytt
+        val temaGammelt = hendelse.temaGammelt
+
+        if (bytterFraEtterlatteTemaTilNoeAnnet(hendelse)) {
+            val journalpost = hentJournalpost(journalpostId)
+            if (journalpostErFerdigstilt(journalpost)) {
+                return
+            }
```

Comment:
Ville fjernet denne for å sikre at oppgaver ikke blir hengende i limbo. 

Eksempel:
- Vi mottar oppgave
- Saksbehandler endrer og ferdigstiller den i Gosys
- Oppgave blir hengende igjen hos oss fordi sjekken ignorerer de ferdigstilte. 

Alternativt må vi få på plass bedre støtte for at sb kan avslutte oppgaver hvor dette har skjedd

Reply:
Enig, fjernet.